pages/home.py

```diff
@@ -75,18 +92,41 @@ def compute_trips_trend(trips_df, date_col):
     )
     return res_df
 
-
+@log_execution_time
 def find_last_get(uuid_list):
-    uuid_list = [UUID(npu) for npu in uuid_list]
-    last_item = list(edb.get_timeseries_db().aggregate([
-        {'$match': {'user_id': {'$in': uuid_list}}},
-        {'$match': {'metadata.key': 'stats/server_api_time'}},
-        {'$match': {'data.name': 'POST_/usercache/get'}},
-        {'$group': {'_id': '$user_id', 'write_ts': {'$max': '$metadata.write_ts'}}},
-    ]))
-    return last_item
+
+    # Do we really need this?
+    # Looks like this takes the most time
+    # uuid_list = [UUID(npu) for npu in uuid_list]
```

Comment:
It's odd that this would be the bottleneck.
Are the entries of `uuid_list` already instances of UUID? Or are they strings?

Reply:
@JGreenlee `uuid_list` is already a string; I don't see why we need to convert it to uuids when we are just querying it.